pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularAgeValidationUpdate.java

```diff
@@ -61,30 +66,59 @@ private void updateValidations(final Handle handle, final ActivityBuilder builde
-        final var activityVersion = handle.attach(JdbiActivityVersion.class)
-                .getActiveVersion(activity.getActivityId())
-                .orElseThrow(() -> new DDPException("Can't fetch latest activity version for " + activity.getActivityCode()));
-
-        handle.attach(SqlHelper.class).deleteValidations(activity.getActivityId());
+        final var revisionId = handle.attach(JdbiActivityVersion.class).findAllVersionsInAscendingOrder(activity.getActivityId())
+                .stream()
+                .findFirst()
+                .map(revision -> handle.attach(JdbiRevision.class).copyStart(revision.getRevId()))
+                .orElseThrow(() -> activeRevisionNotFoundError(activity.getActivityCode(), cfg.getString("study.guid")));
 
         builder.insertValidations(handle,
                 activity.getActivityId(),
                 activity.getActivityCode(),
-                activityVersion.getRevId(),
+                revisionId,
                 List.copyOf(config.getConfigList("validations")));
     }
 
+    private void removeValidations(final Handle handle, final String activityCode, final Config config) {
+        List.copyOf(config.getConfigList("validations")).forEach(validation -> removeValidation(handle, activityCode, validation));
+    }
+
+    private void removeValidation(final Handle handle, final String activityCode, final Config validation) {
+        handle.attach(SqlHelper.class)
+                .removeValidation(activityCode, validation.getString("precondition"), validation.getString("expression"));
+    }
+
+    private DDPException activeRevisionNotFoundError(final String activityCode, final String studyGuid) {
+        return new DDPException(String.format("failed to find an active revision for [activity:%s] in [study:%s]",
+                activityCode,
+                studyGuid));
+    }
+
     private interface SqlHelper extends SqlObject {
-        @SqlUpdate("DELETE FROM activity_validation WHERE study_activity_id = :studyActivityId")
-        void deleteValidations(@Bind("studyActivityId") long studyActivityId);
+        @SqlUpdate("DELETE av "
+                 + "FROM activity_validation av "
+                 + "JOIN study_activity sa ON av.study_activity_id = sa.study_activity_id "
+                 + "WHERE sa.study_activity_code = :activityCode "
+                 + "AND REPLACE(REPLACE(REPLACE(REPLACE(av.expression_text, ' ', ''), CHAR(13), ''), CHAR(10), ''), CHAR(9), '') = "
+                 + "    REPLACE(REPLACE(REPLACE(REPLACE(:expression, ' ', ''), CHAR(13), ''), CHAR(10), ''), CHAR(9), '') "
+                 + "AND REPLACE(REPLACE(REPLACE(REPLACE(av.precondition_text, ' ', ''), CHAR(13), ''), CHAR(10), ''), CHAR(9), '') = "
+                 + "    REPLACE(REPLACE(REPLACE(REPLACE(:precondition, ' ', ''), CHAR(13), ''), CHAR(10), ''), CHAR(9), '') ")
```

Comment:
Can you add a comment to this method detailing why we're performing the string of `REPLACE` calls?

Reply:
Added! I hope the explanation is clear enough :)